src/App.js

```diff
@@ -1,79 +1,104 @@
-import React from 'react'
-import { Experiment, jsPsych } from 'jspsych-react'
-import { tl } from './timelines/main'
-import { MTURK, FIREBASE } from './config/main'
-import './App.css'
-import 'bootstrap/dist/css/bootstrap.css'
-import '@fortawesome/fontawesome-free/css/all.css'
-import { getTurkUniqueId, sleep } from './lib/utils'
-import { rt_categorize_html } from './lib/rt-categorize-html'
-import { addToFirebase, createFirebaseDocument,} from "./firebase.js";
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import "bootstrap/dist/css/bootstrap.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+import Login from "./components/Login";
+import { initParticipant } from "./firebase";
+import JsPsychExperiment from "./components/JsPsychExperiment";
+import { MTURK, IS_ELECTRON, AT_HOME, FIREBASE, PROLIFIC } from "./config/main";
+import { getTurkUniqueId } from "./lib/utils";
 
+function App() {
+  // Variables for login
+  const [loggedIn, setLogin] = useState(false);
+  const [ipcRenderer, setRenderer] = useState(false);
+  const [psiturk, setPsiturk] = useState(false);
+  const [envParticipantId, setEnvParticipantId] = useState("");
+  const [envStudyId, setEnvStudyId] = useState("");
```

Comment:
Similarly here, I don't think this needs to be state. Is there are reason why you've made this state? Will it change throughout the run of the experiment?